pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/DefaultPicoConfigBeanRegistry.java

```diff
@@ -332,66 +329,36 @@ <T, GCB extends GeneratedConfigBean> void loadConfigBeans(io.helidon.config.Conf
                     + config.key().toString());
         }
 
-        ConfigValue<List<io.helidon.config.Config>> nodeList = config.asNodeList();
-        GCB baseConfigBean = maybeLoadBaseConfigBean(config, nodeList, configuredServiceProvider);
-        Map<String, GCB> mapOfInstanceBasedConfig = maybeLoadConfigBeans(nodeList, configuredServiceProvider);
-
-        // validate what we've loaded, to ensure it complies to the meta config info policy
-        if (!metaConfigBeanInfo.repeatable() && !mapOfInstanceBasedConfig.isEmpty()) {
-            throw new ConfigException("Expected to only have a single base, non-repeatable configuration for "
-                                              + configuredServiceProvider.serviceType() + " with config: "
-                                              + config.key().toString());
-        }
-
-        if (baseConfigBean != null) {
+        if (metaConfigBeanInfo.repeatable()) {
+            // this config bean may be more than once - if list, use children, otherwise use child nodes
+            registerChildConfigBeans(config, configuredServiceProvider, metaAttributes);
+        } else {
+            // not a repeatable one, use this node directly, if list, this is a problem
+            if (config.isList()) {
+                throw new ConfigException("Expected to only have a single base, non-repeatable configuration for "
```

Comment:
advise in the exception to use `repeatable=true` on the ConfigBean if they desire repeatability.

Reply:
I did not change the exception text at all in this PR, so this is the original message.

Reply:
Updating it anyway...